site/source/docs/api_reference/vr.h.rst

```diff
@@ -0,0 +1,378 @@
+	running browser.
+
+	:returns: `1` if ready, `0` otherwise.
+	:rtype: int
+
+API Queries
+===========
+
+All of the following functions require :c:func:`emscripten_vr_init` to have been
+called but do not require VR displays and can therefore be called before the return
+value of :c:func:`emscripten_vr_ready` is `true`.
+
+.. c:function:: int emscripten_vr_version_minor()
+
+	Minor version of the WebVR API currently supported by the browser.
+
+	:returns: minor version of WebVR, or `-1` if not supported or API not initialized.
+	:rtype: int
+
+.. c:function:: int emscripten_vr_version_major()
+
+	Major version of the WebVR API currently supported by the browser.
+
+	:returns: major version of WebVR, or `-1` if not supported or API not initialized.
+	:rtype: int
+
+Display Functions
+=================
+
+All of the following functions require :c:func:`emscripten_vr_init` to have been
+called the return value of :c:func:`emscripten_vr_ready` to be `true`.
+
+.. c:function:: int emscripten_vr_count_displays()
+
+	:returns: Number of displays connected.
+	:rtype: int
+
+.. c:function:: VRDisplayHandle emscripten_vr_get_display_handle(int displayIndex)
+
+	:param int displayIndex: index of display (inclusive 0 to exclusive :c:func:`emscripten_vr_count_displays`).
+	:returns: handle for a VR display.
+	:rtype: VRDisplayHandle
+
+.. c:function:: char* emscripten_vr_get_display_name(VRDisplayHandle handle)
+
+	Get a user-readable name which identifies the VR display.
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:returns: name of the VR display or `0 (NULL)` if the handle is invalid.
+	:rtype: char*
+
+.. c:function:: bool emscripten_vr_get_display_connected(VRDisplayHandle handle)
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:returns: `true` if the display is connected, `false` otherwise or when
+		the handle is invalid.
+	:rtype: bool
+
+.. c:function:: bool emscripten_vr_get_display_presenting(VRDisplayHandle handle)
+
+	See also :c:func:`emscripten_vr_request_present`.
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:returns: `true` if the display is currently presenting, `false` otherwise
+		or when the handle is invalid.
+	:rtype: bool
+
+.. c:function:: int emscripten_vr_get_display_capabilities(VRDisplayHandle handle, VRDisplayCapabilities* displayCaps)
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:param VRDisplayCapabilities displayCaps: receives capabilities of the VR display.
+	:returns: |display-function-return-doc|
+	:rtype: bool
+
+.. c:function:: int emscripten_vr_get_eye_parameters(VRDisplayHandle handle, VREye whichEye, VREyeParameters* eyeParams)
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:param VREye whichEye: which eye to query parameters for.
+	:param VREyeParameters eyeParam: receives the parameters for requested eye.
+	:returns: |display-function-return-doc|
+	:rtype: bool
```

Comment:
On all of these success/fail functions, I'd like to see `EMSCRIPTEN_RESULT` being used, though I do notice it would flip the success/fail meaning the other way around and that might confuse, so your judgement call. The one future proofing scenario that I'm thinking here where EMSCRIPTEN_RESULT might be useful is to carry minimal amount of error reasons, e.g. `EMSCRIPTEN_RESULT_NOT_SUPPORTED` if browser does not support WebVR, and `EMSCRIPTEN_RESULT_INVALID_TARGET` if e.g. one lost access to the VR device, or `EMSCRIPTEN_RESULT_INVALID_PARAM` if user passed in something bad, and so on. Also  `EMSCRIPTEN_RESULT_DEFERRED` and/or `EMSCRIPTEN_RESULT_FAILED_NOT_DEFERRED` may be useful when making the API multithreading capable, in the scenarios where we may need to proxy stuff over to the main thread.

Reply:
I agree and I do not shy the effort required to change all the return values. Please see my comment [here](https://github.com/kripken/emscripten/pull/5508#discussion_r143470069) about how to handle the dependency on html5.h.

> though I do notice it would flip the success/fail meaning

Then it is important to do this now before this gets released (and I'm probably the only one who needs to adapt his code)! 👍